src/osx/core/colour.cpp

```diff
@@ -163,27 +163,37 @@ wxColour::wxColour(CGColorRef col)
 
 wxColour::ChannelType wxColour::Red() const
 {
+    wxCHECK_MSG( IsOk(), 0, wxT("invalid colour") );
```

Comment:
I think it's rather unexpected for `wxColour` to be invalid and I don't think it can actually happen in any other port (wxMSW has `m_isInit` field, but will still happily return 0 from `Red()` even when it's set to false).

So even though usually I'm all for adding asserts, I think it would be better to just return 0 for non-initialized colours here too. Otherwise we have a choice between different ports behaving differently or adding asserts to wxMSW and wxGTK and possibly triggering tons of them (quite likely in drawing code, where asserts almost always result in a crash as they're reentered) in the existing code.

Reply:
I don't feel strongly about these asserts, but let me explain the motivation: after Stefan's `NSColor` changes, some incorrect code that previously worked started crashing on NULL access — that's what lead to 4adf620. I figured user code may be affected to and that it might be worth doing.

I didn't know wxMSW doesn't do these checks, to be honest. I copied the asserts pretty much verbatim from wxGTK, which does.

Reply:
OK, sorry, I didn't know that wxGTK already had these asserts. Let's have them here too (although it'd be nice to avoid copying `wxT()` around...), and I guess that we ought to add them to wxMSW too then for consistency.